### ISSUE TITLE
Allow hiding arbitrary columns to improve gappy displays, or all gaps from a particular row

### ISSUE THREAD

**Issue by cmdcolin** [GMOD/react-msaview#25]
This is a nice jalview-ism

User can select arbitrary columns and hide them...useful for gappy alignments

Users can also select a particular row and hide all gaps from that row

**Comment by ihh** [comment]
Still really wish we could animate these sorts of transition...

**Comment by cmdcolin** [comment]
I know, it's unfortunate that I don't have animations and I am not sure it can be bolted onto this app easily. It would be a big undertaking and I am just not an animation expert. I also think, to really get stuff like that right, we probably would need to go webgl. See https://cov2tree.org/ for a great example of webgl large trees. This app can handle the same scale as cov2tree demo there, but the smooth zooming and animation is something I don't think that I can really reproduce in this app. If there is a need for these things, we may need to redesign

**Comment by ihh** [comment]
Yes, I have large tree-envy looking at cov2tree.org. Consensus seems to be we need to go to WebGL for that. If I had to choose, I'd say the stuff you're doing with adding annotations to react-msaview (and better all-round integration) is more important than animations, so I think it's on the right track, but I do think WebGL would be cool.

**Comment by cmdcolin** [comment]
there is an interesting demo from https://genomic.social/@martinsteinegger@mstdn.science/112896963254440965 that hides gappy columns based on a percentage. number of other cool features in that demo but that is a notably cool idea related to this issue

**Comment by cmdcolin** [comment]
the gappyness percentage slider has been implemented now in our code...might be able to close for now
